### ISSUE TITLE
[menu] In Safari, keyboard navigation fails after clicking on submenu

### ISSUE THREAD

**Issue by sorvell** [material-components/material-web#5013]
### What is affected?

Component

### Description

1. On Safari, go to the menu demo: https://material-web.dev/components/menu/ and expand the interactive demo
2. click on the "Menu with submenus" `Open Menu` button.
3. click on one of the items in the menu with a submenu/
4. now use arrow keys to navigate the menu.

Expected: arrow keys can navigate the menu.
Actual: the up/down keys do nothing and the left/right seem to work unreliably.

### Reproduction

https://material-web.dev/components/menu/

### Workaround

N/A

### Is this a regression?

No or unsure. This never worked, or I haven't tried before.

### Affected versions

Failing in 1.0

### Browser/OS/Node environment

Browser: Safari 16.5.2
